bundle/View/Provider/Configured.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Netgen\Bundle\EzPlatformSiteApiBundle\View\Provider;
+
+use eZ\Publish\Core\MVC\Symfony\Matcher\MatcherFactoryInterface;
+use eZ\Publish\Core\MVC\Symfony\View\ContentView as CoreContentView;
+use eZ\Publish\Core\MVC\Symfony\View\View;
+use eZ\Publish\Core\MVC\Symfony\View\ViewProvider;
+use Netgen\Bundle\EzPlatformSiteApiBundle\DependencyInjection\Configuration\Parser\ContentView as ContentViewParser;
+use Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollectionMapper;
+use Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView;
+use Symfony\Component\HttpKernel\Controller\ControllerReference;
+
+/**
+ * A kind of a plugin to the Configurator, uses view configuration.
+ *
+ * @see \eZ\Publish\Core\MVC\Symfony\View\Configurator\ViewProvider
+ */
+class Configured implements ViewProvider
+{
+    /**
+     * @var \eZ\Publish\Core\MVC\Symfony\Matcher\MatcherFactoryInterface
+     */
+    protected $matcherFactory;
+
+    /**
+     * @var \Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollectionMapper
+     */
+    private $queryCollectionMapper;
+
+    /**
+     * @param \eZ\Publish\Core\MVC\Symfony\Matcher\MatcherFactoryInterface $matcherFactory
+     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollectionMapper $queryCollectionMapper
+     */
+    public function __construct(
+        MatcherFactoryInterface $matcherFactory,
+        QueryCollectionMapper $queryCollectionMapper
+    ) {
+        $this->matcherFactory = $matcherFactory;
+        $this->queryCollectionMapper = $queryCollectionMapper;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * Returns view as a data transfer object.
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentType
+     */
+    public function getView(View $view)
+    {
+        if (($configHash = $this->matcherFactory->match($view)) === null) {
+            return null;
+        }
+
+        // This service is dispatched by the configured view class, so this assertion should be safe
+        assert($view instanceof ContentView);
```

Comment:
Still a leftover?

Reply:
Whoops, removed in https://github.com/netgen/ezplatform-site-api/pull/70/commits/abac8b50c33b18f5d56c643ba38786a857ed66dd